### PR TITLE
don't error on /issues - allow GitHub URL as argument in URL

### LINKED ISSUE #1: pasting in url with `/issues` does not work
if i paste in a url with `/issues` at the end
i get no results

### LINKED ISSUE #2: include user-specified path as an argument in the website url itself
if i could add a user-specified path as an argument on the url itself, i could always hit "one random one"

so like

`https://tobycm.github.io/random-issue-picker/?https://github.com/esaruoho/org.lackluster.Paketti.xrnx/issues`  and it would dump out a result.

### PR DESCRIPTION
**Pull request by esaruoho**
Hi @tobycm 
closes #1 
closes #2 

so now it is possible to do this: (add a /issues/ URL and still get a result)
Screenshot 2024-10-08 at 13 02 53

and this: (have `?githuburl` in the url and thus be able to click on "pick one" to get a result.
Screenshot 2024-10-08 at 13 03 03

with this
```
file:///Users/esaruoho/work/random-issue-picker/index.html?https://github.com/esaruoho/org.lackluster.Paketti.xrnx/issues
```

so it is possible to automatically have a URL and then press the button and get a random one. thanks.

### CODE REVIEW

**Comment by tobycm** [comment]
Thanks, I will take a look and merge it later.

**Comment by esaruoho** [comment]
thanks @tobycm ! Looking forward to it :+1:

File: index.html [hunk 1]
```diff
@@ -80,10 +80,14 @@ <h1>Random Issue Picker</h1>
        * Parses the URL of a GitHub repository.
        *
        * @param {string} url The URL of the GitHub repository.
-       * @returns {string} The name of the repository.
+       * @returns {string | null} The name of the repository.
```
Comment (tobycm) [inline index.html:83]: nice

Comment (tobycm) [inline index.html:83]: nice

File: index.html [hunk 2]
```diff
@@ -139,7 +144,8 @@ <h1>Random Issue Picker</h1>
           const link = response.headers.get("Link");
           if (!link) return 1;
 
-          return link.match(/page=(\d+)>; rel="last"/)[1];
+          const match = link.match(/page=(\d+)>; rel="last"/);
+          return match ? parseInt(match[1], 10) : 1;
```
Comment (tobycm) [inline index.html]: I guess we can throw this in a function

File: index.html [hunk 3]
```diff
@@ -174,8 +180,9 @@ <h1>Random Issue Picker</h1>
         else issues = [];
 
         try {
-          const response = await fetch(
-            `https://api.github.com/repos/${repoInfo.name}/issues?per_page=1&page=${Math.floor(Math.random() * repoInfo.issues) + 1}`
+          const randomPage = Math.floor(Math.random() * repoInfo.issues) + 1;
+          let response = await fetch(
+            `https://api.github.com/repos/${repoInfo.name}/issues?per_page=1&page=${randomPage}`
```
Comment (tobycm) [inline index.html]: my bad for one line XD

**Comment by tobycm** [comment]
I changed the url query stuff a bit
`/?url=https://github.com/esaruoho/org.lackluster.Paketti.xrnx/issues&fetch&open`
`url` set to repo url
`fetch` to fetch an issue
`open` to open in new tab

**Comment by esaruoho** [comment]
Thanks, wonderful and clever changes and additions. Did you deploy it to the url?

**Comment by tobycm** [comment]
Yes, CloudFlare Pages CI/CD :heart:

**Pull request merged**